src/building/compiler-documenting.md

```diff
@@ -1,55 +1,50 @@
 # Building documentation
 
-You might want to build documentation of the various components
-available like the standard library. There’s two ways to go about this.
-You can run rustdoc directly on the file to make sure the HTML is
-correct, which is fast. Alternatively, you can build the documentation
-as part of the build process through `x.py`. Both are viable methods
-since documentation is more about the content.
+This chapter describes how to build documentation of toolchain components,
+either in whole or individually.
 
-## Document everything
+- Document everything
 
-This uses the beta rustdoc, which usually but not always has the same output
-as stage 1 rustdoc.
+  This uses `rustdoc` from the beta toolchain,
+  so will produce (slightly) different output to stage 1 rustdoc,
+  as `rustdoc` is under active development:
 
-```bash
-./x.py doc
-```
+  ```bash
+  ./x.py doc
+  ```
 
-## If you want to be sure that the links behave the same as on CI
+  If you want to be sure the documentation looks the same as on CI:
 
-```bash
-./x.py doc --stage 1
-```
+  ```bash
+  ./x.py doc --stage 1
+  ```
 
-First the compiler and rustdoc get built to make sure everything is okay
-and then it documents the files.
+  First,
+  the compiler and rustdoc get built to make sure everything is okay,
+  and then it documents the files.
```

Comment:
hmm, this sentence is saying a lot of complicated things in very few words.
- rustdoc doesn't check that programs are valid, so you have to run rustc directly to make sure the program compiles
- rustdoc uses the built metadata to document the compiler (it's very unclear in this sentence whether "it" refers to rustdoc or bootstrap).

unless it's talking about stage1, in which case it's just completely wrong - the reason you have to build rustdoc is so you can run it, because it might be different from the beta rustdoc.

Reply:
(this is also one of the few cases where `--keep-stage 0` is sound - if you've only modified documentation, it can't affect the build artifacts, so it's ok to only rebuild the documentation and not rustdoc itself.)

Reply:
not sure how much detail to provide, but please have a look at the small change I made